{{cookiecutter.project_name}}/contracts/NFT.vy

```diff
@@ -108,6 +116,14 @@ isApprovedForAll: public(HashMap[address, HashMap[address, bool]])
 
 # @dev Mapping from NFT ID to approved address.
 idToApprovals: public(HashMap[uint256, address])
+{%- if cookiecutter.force_royalties == 'y' %}
+# @dev the last balance of the smart contract that stores the royalties of the contract creator
+# this balance is reset to 0 the moment the creator withdraws royalties
+lastBalance: uint256
+
+# @dev we check this value to make sure royalties have been paid
+royaltyAmount: uint256
```

Comment:
Would suggest `minRoyaltyAmount`

Reply:
Also, add a note here to explore means of setting `minRoyaltyAmount` based on floor price oracle

Reply:
Also also, need to set this in `__init__` to a nonzero value